### PR TITLE
PADV-626 & 627 - Add Filters component and handleStudents for Students Page.

### PR DESCRIPTION
**Pull request by anfbermudezme**
## Tickets

https://agile-jira.pearson.com/browse/PADV-626
https://agile-jira.pearson.com/browse/PADV-627

## Description

Following the MVP requirements we need to filter by Class Title, Instructor Email and Name and also we need to add a button to revoke a pending user, that is, it is in CourseEnrollmentAllowed and has not been registered on the platform.

## Changes made

- [x] Add Filters Component and integrate it in StudentsPage,
- [x] Add Components from Paragon to open Modal for Filters.

image

- [x] Add handleEnrollments action API.
- [x] Add required columns to revoke pending enrollments.

image

- [x] Add handleEnrollments in Students Page to revoke pending enrollments.

image

- [x] Add respective test.

## How to test

- Clone the Institution Portal MFE
- Run npm install.
- Run npm run start

### CODE REVIEW

File: src/features/Students/StudentsPage/index.jsx [hunk 1]
```diff
@@ -58,35 +124,103 @@ const StudentsPage = () => {
...
+        <OverlayTrigger
+          placement="top"
+          overlay={<Tooltip variant="light">Clean filters</Tooltip>}
+        >
+          <IconButton icon={faTrash} alt="filter" onClick={handleCleanFilters} variant="secondary" />
+        </OverlayTrigger>
+      </ActionRow>
       <StudentsTable
-        data={state.data}
+        data={state.data || []}
```
Comment (sergivalero20) [inline src/features/Students/StudentsPage/index.jsx:193]: @anfbermudezme what's the reason to include this condition `|| []`? data is already declared as an empty array in initialState.

File: src/features/Students/StudentsTable/columns.jsx [hunk 2]
```diff
@@ -26,8 +32,23 @@ const getColumns = () => [
   {
     Header: 'Created',
     accessor: 'created',
-    Cell: ({ row }) => new Date(row.values.created).toUTCString(),
```
Comment (sergivalero20) [inline src/features/Students/StudentsTable/columns.jsx:29]: @anfbermudezme why did you remove this?

File: src/features/Students/StudentsFilters/index.jsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+          placeholder="Enter Instructor"
+          value={filters.instructor}
+          onChange={handleInputChange}
+          className="mb-3"
+        />
+        <Form.Control
+          type="text"
+          floatingLabel="Class Name"
+          name="ccx_name"
+          placeholder="Enter Class Name"
```
[the remark this answers is not in the document]
Comment (anfbermudezme) [reply src/features/Students/StudentsFilters/index.jsx:52]: You are right, I think the Class Name is clearer, because in the platform it is known as course name or ccx name. So I changed the name in the column, do you agree?

[the remark this answers is not in the document]
Comment (sergivalero20) [reply src/features/Students/StudentsFilters/index.jsx:52]: @anfbermudezme yes, It's ok. Please, update screenshot to see that changes.

[the remark this answers is not in the document]
Comment (anfbermudezme) [reply src/features/Students/StudentsFilters/index.jsx:52]: Done

[the remark this answers is not in the document]
Comment (sergivalero20) [reply src/features/Students/StudentsFilters/index.jsx:52]: @anfbermudezme you're missing update the first screenshot.

**Review by sergivalero20 — APPROVED** [review]

**Review by Squirrel18 — APPROVED** [review]
